### PR TITLE
Document improved USB HID handling on Qubes 4.1 

### PR DESCRIPTION
**Pull request by marmarek**
1. Describe setup with separate USB controller.
2. Add info about built-in USBGuard support.

https://github.com/QubesOS/qubes-issues/issues/6959

### CODE REVIEW

**Comment by andrewdavidwong** [comment]
Wow, this page is a huge mess (not because of your changes; it already was one). There are haphazard, inconsistent instructions to accomplish the same tasks scattered throughout the document, outdated information, repetitive references and notes, and no clear organization. This is basically the documentation version of spaghetti code. Even an advanced user is likely to have difficulty understanding this page, and if they actually tried following the instructions, they would often find that they just performed a bunch of unnecessary steps because some new automatic method is documented _after_ the old manual method. I think this page needs a complete overhaul. I can't sensibly edit just the added content alone. However, as a laptop user, I don't have first-hand experience with multi-USB-controller configurations, USB keyboards, etc., so it will be challenging to reorganize content that I don't fully understand. Still, I'll try my best.

**Comment by andrewdavidwong** [comment]
PipelineRetry

**Review by andrewdavidwong — APPROVED** [review]

**Comment by andrewdavidwong** [comment]
Revamp pushed and CI errors fixed. But it won't let me request a review from you, @marmarek. Your name is not in the list. I guess because you're the one who opened the PR, and GitHub assumes no one would ever need to review their own PR (even after someone else pushes changes to it). So I'm just assigning instead.

**Comment by andrewdavidwong** [comment]
Reminder: Even once it's approved, this PR won't be merged until the 4.1 stable release.

**Comment by andrewdavidwong** [comment]
I just realized that, when I revised this, I forgot that there will be (is now) a six-month period where 4.0 is still supported alongside 4.1, so it was wrong of me to remove references to 4.1 (though it was also wrong to insert them in-text rather than create [separate sections](https://www.qubes-os.org/doc/documentation-style-guide/#release-specific-documentation)).

**Comment by andrewdavidwong** [comment]
If you can point out the few instances (there should only be a few, right?) where 4.0 and 4.1 differs in this (revised) version, I can add the separate-sections-for-each-release thing (as long as I can figure out from the commit history which content belongs in which). If nothing happens in six months, this can just be merged as-is after passing review.

File: user/advanced-topics/usb-qubes.md [hunk 1]
```diff
@@ -95,26 +102,67 @@ sys-usb dom0 allow
 
 (Change `sys-usb` to your desired USB qube.)
 
-You can now use your USB keyboard to login and for LUKS decryption during boot.
+You can now use your USB keyboard to log in and for LUKS decryption during
+boot.
```
Comment (marmarek) [inline user/advanced-topics/usb-qubes.md:106]: The above does not cover LUKS passphrase prompt (sys-usb is not running at this point yet). This one is covered by removing `rd.qubes.hide_all_usb` option from the kernel cmdline.

File: user/advanced-topics/usb-qubes.md [hunk 2]
```diff
@@ -95,26 +102,67 @@ sys-usb dom0 allow
...
 
-For a confirmation dialog each time the USB keyboard is connected, *which will effectively disable your USB keyboard for login and LUKS decryption*, change this line to:
+You can set up your system so that there's a confirmation prompt each time the
+USB keyboard is connected. However, this will effectively disable your USB
+keyboard for login and LUKS decryption, so **don't do this if you want to
+unlock your device with a USB keyboard!** If you're sure you wish to proceed,
+change the previous line to:
```
Comment (marmarek) [inline user/advanced-topics/usb-qubes.md:112]: No LUKS decryption here either.

File: user/advanced-topics/usb-qubes.md [hunk 3]
```diff
@@ -95,26 +102,67 @@ sys-usb dom0 allow
...
-If you want to attach the USB mouse automatically anyway, you have to edit the `qubes.InputMouse` policy file in dom0, located at:
+If you wish to use a USB keyboard to enter your LUKS passphrase, you cannot
+[hide its USB controller from dom0](#how-to-hide-usb-controllers-from-dom0). If
+you've already hidden that USB controller from dom0, you must revert the
+procedure by removing the `rd.qubes.hide_all_usb` option and employ an
+alternative strategy for protecting your system by physically disconnecting
+other devices during startup. You should also add the
+`usbcore.authorized_default=0` option, which prevents the initialization of
+non-input devices. (Qubes ships with a USBGuard configuration that allows only
+input devices when `usbcore.authorized_default=0` is set.)
```
Comment (marmarek) [inline user/advanced-topics/usb-qubes.md:126]: The last part about `usbcore.authorized_default=0` is R4.1-only.

File: user/advanced-topics/usb-qubes.md [hunk 4]
```diff
@@ -95,26 +102,67 @@ sys-usb dom0 allow
...
+
+
+### How to enable a USB keyboard on a separate USB controller
```
Comment (marmarek) [inline user/advanced-topics/usb-qubes.md:129]: This section is R4.1-only. You can achieve subset of it manually on R4.0, but it's a bit clumsy so I don't think it's worth describing here.

File: user/advanced-topics/usb-qubes.md [hunk 5]
```diff
@@ -126,77 +174,122 @@ The first line should read similar to:
...
+2. In Qubes Manager, right-click on the USB qube and select "Remove VM."
+3. Open the file `/etc/default/grub` in dom0.
+4. Find the line(s) that begins with `GRUB_CMDLINE_LINUX`.
+5. If `rd.qubes.hide_all_usb` appears anywhere in those lines, remove it.
+6. Save and close the file.
+7. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
+8. Reboot.
 
-* EFI
+### EFI
```
Comment (marmarek) [inline user/advanced-topics/usb-qubes.md:287]: Maybe worth clarifying it's about R4.0 only? Because on R4.1 we have Grub on EFI too.

**Comment by andrewdavidwong** [comment]
All fixed. Thanks! Merging now.

**Pull request merged**